src/transformers/modeling_utils.py

```diff
@@ -1308,6 +1313,9 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
                         )
                 dtype_orig = cls._set_default_torch_dtype(torch_dtype)
 
+            if low_cpu_mem_usage:
+                loaded_state_dict_keys = cls._load_state_dict_into_model_low_mem_p1(state_dict)
```

Comment:
It isn't clear to me at first glance that p1 means part1. Also that part is short enough that it doesn't necessarily need its own function, but if you do make one, please name it something different: it doesn't load the state dict into the model, just extract the keys, so ` cls._extract_keys_from_state_dict` would be a better name.

For the caching, we can implement it in a followup PR if we see a usage for this feature.

Reply:
I agree it's not the best naming, this was just an attempt to remove any noise that doesn't help the reader to quickly understand what's going on when they don't use this option.

One proposal is to have the 2 parts named:
1. `load_state_dict_into_model_low_mem_prepare`
2. `load_state_dict_into_model_low_mem`

I used the same prefix, so that it's clear they are 2 parts of the same functionality.

Note that it doesn't just extract the keys, but drops `state_dict`. But otherwise your suggestion would have been perfect.

Reply:
I would still use a more descriptive name as what is now only used for low memory loading may very well be re-used in the future for something else, and then a name that doesn't a have a direct link with the functionality won't work. It is crystal clear from the code that this is only executed for `low_cpu_mem_usage` from the test before.

Reply:
I also think `retrieve_keys_from_state_dict` or `extract_keys_from_state_dict` are good names.

Reply:
As I tried to explain earlier, this helper function is not doing just  `retrieve_keys_from_state_dict` - it deletes `state_dict`!

Let's resolve this by reverting the proposed change, I will just revert those 2 lines to be just that and require no function.

Reply:
So now it's back to the simple:

```
            if low_cpu_mem_usage:
                # save the keys
                loaded_state_dict_keys = [k for k in state_dict.keys()]
                del state_dict  # free CPU memory - will reload again later
```